File: icpc/2023/E/extravaganza_edwin_AC.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, tcnt;
    string S;
    cin>>tcnt;
    while (tcnt!=0) {
        cin>>N; cin>>S;
        int open=0, sol=0; 
        for(int i=0; i<S.size();i++)
            if (S[i]=='(') open++;
            else if (open>0) {
                open--;
                sol+=2;
            }
        cout<<sol<<"\n";
        tcnt--;
    }
    return 0;
}

File: icpc/2023/E/extravaganza_edwin_AC.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class extravaganza_edwin_AC {

    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuffer buf = new StringBuffer();

    public static void main(String[] args) throws Exception {
        int N;
        char[] S;
        String ln = in.readLine();
        int tcnt=Integer.parseInt(ln);
        while (tcnt!=0) {
        // while (ln!=null && !ln.trim().equals("")) {
            ln = in.readLine();
            st = new StringTokenizer(ln);
            N = Integer.parseInt(st.nextToken());
            S = st.nextToken().toCharArray();
            int open=0, sol=0;
            for(int i=0; i<S.length;i++)
                if (S[i]=='(') open++;
                else if (open>0) {
                    open--;
                    sol+=2;
                }
            buf.append(sol+"\n");
            tcnt--;
        }
        System.out.print(buf);
    }
}
